File: backend/app/schemas/cve.py

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class CVEBase(BaseModel):
    id: str
    description: str
    severity: str
    cvss: float
    epss: float
    kev: bool
    published: datetime
    modified: datetime


class CVEResponse(CVEBase):
    model_config = ConfigDict(from_attributes=True)


class CVEListResponse(BaseModel):
    items: list[CVEResponse]
    total: int

File: frontend/lib/services/cve.ts
import { api } from "@/lib/api";

export interface CVE {
  id: string;
  description: string;
  cvss: number;
  severity: string;
  epss: number;
  epss_percentile?: number;
  kev: boolean;
  published: string;
  modified: string;
  mitre?: unknown;
}

export interface CVEQuery {
  limit?: number;
  start_index?: number;
  severity?: string;
  keyword?: string;
}

export interface CVEResponse {
  items: CVE[];
  total: number;
  count: number;
  limit: number;
  start_index: number;
}

export async function getCVEs(
  params: CVEQuery = {}
): Promise<CVEResponse> {
  const { data } = await api.get<CVEResponse>("/cves/", {
    params: {
      limit: params.limit ?? 20,
      start_index: params.start_index ?? 0,
      ...(params.severity ? { severity: params.severity } : {}),
      ...(params.keyword ? { keyword: params.keyword } : {}),
    },
  });

  return data;
}
